src/WidgetBase.ts

```diff
@@ -335,20 +335,24 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 	 */
 	protected addDecorator(decoratorKey: string, value: any): void {
 		value = Array.isArray(value) ? value : [ value ];
-
-		let decoratorList = decoratorMap.get(this.constructor);
-		if (!decoratorList) {
-			decoratorList = new Map<string, any[]>();
-			decoratorMap.set(this.constructor, decoratorList);
+		if (this._properties) {
+			const decorators = this._decoratorCache.get(decoratorKey) || [];
+			this._decoratorCache.set(decoratorKey, [ ...decorators, ...value ]);
```

Comment:
Since `_decoratorCache` is populated on the first call to `getDecorator`, I think maybe that calling `addDecorator` and then `getDecorator` will _only_ get the one you just put in the cache, and not any of the others that are on the class level (because the cache is set so it won't build the decorator list). Am I crazy?